src/AppBundle/Controller/UserController.php

```diff
@@ -60,20 +61,19 @@ class UserController extends BaseController
      */
     public function createUserAction(CreateUserDTO $dto)
     {
-        /** @var \AppBundle\Model\User\Registration\TwoStepRegistrationApproach $registrator */
-        $registrator = $this->get('app.user.registration');
+        $this->handle($dto);
 
-        $result = $registrator->registration($dto);
-        if (!$result->isValid()) {
-            $response = ['errors' => $this->sortViolationMessagesByPropertyPath($result->getViolations())];
-            if (!empty($result->getSuggestions())) {
-                $response['name_suggestions'] = $result->getSuggestions();
+        if (($info = $dto->getInfo()) && !$info->isValid()) {
+            $response    = ['errors' => $this->sortViolationMessagesByPropertyPath($info->violationList)];
+            $suggestions = $info->getExtraValue(NameSuggestionMiddleware::SUGGESTIONS, true);
```

Comment:
this constant should be moved into the `Command` object since middlewares shouldn't be used for any purpose in another API (managed private by the command bus)